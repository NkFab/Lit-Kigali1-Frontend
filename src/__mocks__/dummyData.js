module.exports = {
  signupUser: {
    username: 'test',
    email: 'test@email.com',
    password: 'test@test',
    firstName: 'test',
    lastName: 'test',
  },
  articleData: {
    id: 'f9a4731d-7dea-4a9a-9b7d-8766d651c202',
    userId: 'dfef16f9-11a7-4eae-9ba0-7038c6ccaa73',
    slug: 'new-article-1',
    title: 'new article one',
    description: 'new article',
    body:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla faucibus ipsum non metus finibus ultricies. Donec ac auctor dui, sed fringilla est. Duis et pellentesque nisl, a gravida felis. Ut tempor felis id dignissim congue. Nunc blandit nunc sit amet dui pharetra, quis porttitor sem ullamcorper. Suspendisse faucibus imperdiet lacinia.',
    status: 'published',
    readingTime: '1 min',
    tagList: ['test', 'article'],
  },
  articleReducerState: {
    articles: [],
  },
  ratingData: {
    id: '1a322238-dadf-406c-8252-df926958622e',
    userId: 'd018c3b5-13c7-41c0-8b2c-4ec1cb6b21da',
    articleId: 'f9a4731d-7dea-4a9a-9b7d-8766d651c202',
    state: null,
    rating: 3,
    createdAt: '2019-04-19T14:14:09.096Z',
    updatedAt: '2019-04-19T17:50:48.826Z',
    author: {
      username: 'username',
      firstName: 'First Name',
      lastName: 'Last Name',
      image: null,
    },
    url: '/articles/article-slug/rating',
  },
  draftjsBody: [
    {
      key: 'dhjdh',
      text:
        'orem ipsum dolor sit amet, consectetur adipiscing elit. Nulla egestas justo arcu, sit amet eleifend velit venenatis et. Curabitur vel ipsum est. Phasellus iaculis arcu vel ligula faucibus laoree',
    },
    {
      key: 'loetvbd',
      text:
        'orem ipsum dolor sit amet, consectetur adipiscing elit. Nulla egestas justo arcu, sit amet eleifend velit venenatis et. Curabitur vel ipsum est. Phasellus iaculis arcu vel ligula faucibus laoree',
    },
    {
      key: 'ottsb',
      text:
        'orem ipsum dolor sit amet, consectetur adipiscing elit. Nulla egestas justo arcu, sit amet eleifend velit venenatis et. Curabitur vel ipsum est. Phasellus iaculis arcu vel ligula faucibus laoree',
    },
    {
      key: 'hsdghsd',
      text:
        'Maecenas a tellus sit amet metus congue dignissim at et ante. Etiam mollis sagittis elit sit amet vestibulum. In hac habitasse platea dictumst. Aliquam ut varius ante',
    },
  ],
  urlValue:
    'https://cdn.shopify.com/s/files/1/0550/6737/products/ORANGEWOOD_MANHATTAN_SPRUCE_BEGINNER_DREADNAUGHT_ACOUSTIC_GUITAR-1_2000x.png?v=1524595098',
  jsonFormat:
    '{"key":"dhjdh","text":"orem ipsum dolor sit amet, consectetur adipiscing elit. Nulla egestas justo arcu, sit amet eleifend velit venenatis et. Curabitur vel ipsum est. Phasellus iaculis arcu vel ligula faucibus laoree"}',
  entityMap1: {
    0: {
      type: 'LINK',
      mutability: 'MUTABLE',
      data: {
        url: 'https://www.lipsum.com/feed/html',
      },
    },
    1: {
      type: 'image',
      mutability: 'IMMUTABLE',
      data: {
        src:
          'https://res.cloudinary.com/litkigaliauthorshaven/image/upload/v1556089726/beach-exotic-holiday-248797_fegfqb.jpg',
      },
    },
    2: {
      type: 'LINK',
      mutability: 'MUTABLE',
      data: {
        url: 'https://www.lipsum.com/feed/html',
      },
    },
    3: {
      type: 'LINK',
      mutability: 'MUTABLE',
      data: {
        url: 'https://www.lipsum.com/feed/html',
      },
    },
    4: {
      type: 'video',
      mutability: 'IMMUTABLE',
      data: {
        src:
          'https://res.cloudinary.com/litkigaliauthorshaven/video/upload/v1556089741/BirdNoSound_1_klovmd.mp4',
      },
    },
  },
  entityMap2: {
    0: {
      type: 'LINK',
      mutability: 'MUTABLE',
      data: {
        url: 'https://www.lipsum.com/feed/html',
      },
    },
    2: {
      type: 'LINK',
      mutability: 'MUTABLE',
      data: {
        url: 'https://www.lipsum.com/feed/html',
      },
    },
    3: {
      type: 'LINK',
      mutability: 'MUTABLE',
      data: {
        url: 'https://www.lipsum.com/feed/html',
      },
    },
    4: {
      type: 'video',
      mutability: 'IMMUTABLE',
      data: {
        src:
          'https://res.cloudinary.com/litkigaliauthorshaven/video/upload/v1556089741/BirdNoSound_1_klovmd.mp4',
      },
    },
  },
  file: {
    lastModified: 1553947426997,
    lastModifiedDate: new Date(),
    name: 'beach-exotic-holiday-248797.jpg',
    size: 522195,
    type: 'image/jpeg',
    webkitRelativePath: '',
  },
  articleDataDraft: {
    id: 'f9a4731d-7dea-4a9a-9b7d-8766d651c202',
    userId: 'dfef16f9-11a7-4eae-9ba0-7038c6ccaa73',
    slug: 'new-article-1',
    title: 'new article one',
    description: 'The CEOs of YouTube and Twitter are being',
    body:
      '{"blocks":[{"key":"crkve","text":"The CEOs of YouTube and Twitter are being far more cautious in addressing the problems of their platforms than they were in building them","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
    status: 'published',
    readingTime: '1 min',
    tagList: ['test', 'article'],
  },
  article: {
    title: 'hello world',
    body: 'ejej lkejke lkejde eljfe kefnef efnefve lkenfefenf kefnerjfnern elkfnrfernflef elkfnejfernf lefnerfe kefnrefne klekfnerfefnlef ekfnefnerel nefefnef efnefklnef kenfefnef kenferfn',
    tagList: [],
  },
};
