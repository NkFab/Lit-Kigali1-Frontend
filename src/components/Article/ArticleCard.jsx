import React, { Component } from 'react';
import { Link } from 'react-dom';
import { PropTypes } from 'prop-types';

class ArticleCard extends Component {
  navigateTo = () => {
    const { history, url } = this.props;
    if (history && url) {
      history.push(url);
    }
  };

  render() {
    const { article, classes } = this.props;
    return (
      <div
        role="presentation"
        className={`article-card ${classes}`}
        key={article.slug}
        onClick={this.navigateTo}
      >
        <div
          className="article-card__image"
          style={{ backgroundImage: `url("${article.cover}")` }}
        />
        <div className="article-card__content">
          <h3 className="article-card__content-title">{article.title}</h3>
          <div className="article-card__content-text">{article.body}</div>
          <div className="article-card__content-meta">
            <span>{article.readingTime}</span>
            <span>
              <i className="fa fa-eye" />
              200
            </span>
            <span>
              <i className="fa fa-thumbs-up" />
              15k
            </span>
          </div>
        </div>
      </div>
    );
  }
}

ArticleCard.propTypes = {
  article: PropTypes.object.isRequired,
  classes: PropTypes.string,
  url: PropTypes.string,
  history: PropTypes.object,
};

ArticleCard.defaultProps = {
  classes: '',
  url: '',
<<<<<<< HEAD
  history: {},
=======
  history: '',
>>>>>>> feat: complete view all articles
};

export default ArticleCard;
