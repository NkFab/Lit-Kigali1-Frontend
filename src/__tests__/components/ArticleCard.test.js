import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import ArticleCard from '../../components/Article/ArticleCard';
import { articleData, signupUser } from '../../__mocks__/dummyData';

let wrapper;
const props = {
  loading: true,
  article: {
    ...articleData,
    author: signupUser,
  },
  url: '/articles/article-slug',
  history: { push: jest.fn() },
};

describe('<ArticleCard />', () => {
  beforeEach(() => {
    wrapper = mount(<ArticleCard {...props} />);
  });
  test('should render the <ArticleCard />', () => {
    const renderedValue = renderer.create(<ArticleCard {...props} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  test("should render the `article's body`", () => {
    const newProps = props;
    newProps.article.desciption = undefined;
    wrapper = mount(<ArticleCard {...newProps} />);
    expect(wrapper.props().article.desciption).toBeUndefined();
  });

  describe('when clicking on navigate button', () => {
    test('should call navigateTo instance function', () => {
      wrapper.find('div[data-el="main-container"]').simulate('click');
      expect(props.history.push).toHaveBeenCalled();
    });

    test('should call toProfile instance function', () => {
      wrapper.find('div[data-el="username"]').simulate('click');
      expect(props.history.push).toHaveBeenCalled();
    });
  });
});
