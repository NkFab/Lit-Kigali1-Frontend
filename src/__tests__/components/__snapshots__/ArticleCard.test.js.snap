// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCard /> should render the <ArticleCard /> 1`] = `ShallowWrapper {}`;
